src/com/sony/qcrilam/BootReceiver.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2018-2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Rewritten in Kotlin by Pavel Dubrova <pashadubrova@gmail.com>
+ */
+
+package com.sony.qcrilam
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+
+import kotlin.jvm.internal.Intrinsics
+
+class BootReceiver : BroadcastReceiver() {
+    private var TAG = "QcRilAm-BootReceiver"
+
+    override fun onReceive(context: Context, intent: Intent?) {
+        Intrinsics.checkParameterIsNotNull(context, "context");
```

Comment:
@bartcubbins Apologies, you don't need this.

I pasted the output of the decompiler just to show what it does :see_no_evil: